src/TwitchLib.Communication/Clients/AClientBaseTyped.cs

```diff
@@ -0,0 +1,93 @@
+﻿
+
+using System;
+
+using Microsoft.Extensions.Logging;
+
+using TwitchLib.Communication.Extensions;
+using TwitchLib.Communication.Interfaces;
+
+namespace TwitchLib.Communication.Clients
+{
+
+    /// <summary>
+    ///     this is a generic "Overlay",
+    ///     to get rid of the <see cref="System.Type"/>-Parameter in <see cref="AClientBase"/>
+    /// </summary>
+    /// <typeparam name="T">
+    ///     should be one of the following
+    ///     <list>
+    ///         <item>
+    ///             <see cref="System.Net.Sockets.TcpClient"/>
+    ///         </item>
+    ///         <item>
+    ///             <see cref="System.Net.WebSockets.ClientWebSocket"/>
+    ///         </item>
+    ///     </list>
+    /// </typeparam>
+    public abstract class AClientBaseTyped<T> : AClientBase where T : IDisposable
```

Comment:
This class seems like unnecessary overhead. I believe you could move everything to `AClientBase`, and make it generic with `IDisposable` restriction.

Reply:
OK. I am going to adopt it.

Reply:
I think the code will be more simple like that. You will get rid of that abstract SetSpecificClient method.